include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -389,7 +389,8 @@ struct __gen_set_mask
 
         std::size_t __nb = __set_b.size();
 
-        const auto& __val_a_proj = std::invoke(__proj1, __set_a[__id]);
+        auto&& __val_a = __set_a[__id];
```

Comment:
```suggestion
        // Extend the lifetime of a possibly temporary value before passing it to the projection
        // which might return a reference to its argument (e.g., std::identity)
        auto&& __val_a = __set_a[__id];
```
It makes sense to repeat the comment in other affected functions as well.